logstash-core/src/main/java/org/logstash/instrument/metrics/Metric.java

```diff
@@ -88,4 +59,16 @@ default String type() {
         return getType().asString();
     }
 
+    /**
+     * Determine if this metric has a value explicitly set.
+     * @return true if this metric has been set to a specific value, false if it is the default state
+     */
+    boolean isDirty();
```

Comment:
I _think_ it is here to stay. 

I guess it depends on if we want to continue serializing only that which has been set, or just output all known values, including the default zeros and nulls. 

The existing code base is a bit inconsistent since in some places it initializes some metrics to zero or null (which flips the dirty flag), but not others metrics. I haven't found much of a pattern to what gets initialized or not.